source/adios2/engine/sst/SstWriter.cpp

```diff
@@ -158,8 +158,7 @@ void SstWriter::EndStep()
         m_BP3Serializer->AggregateCollectiveMetadata(
             m_MPIComm, m_BP3Serializer->m_Metadata, true);
         BP3DataBlock *newblock = new BP3DataBlock;
-        newblock->metadata.DataSize =
-            m_BP3Serializer->m_Metadata.m_Buffer.size();
+        newblock->metadata.DataSize = m_BP3Serializer->m_Metadata.m_Position;
         newblock->metadata.block = m_BP3Serializer->m_Metadata.m_Buffer.data();
```

Comment:
So this one is actually not the buffer I meant ;) I wasn't talking about the metadata buffer but about the actual data buffer. I haven't looked at the metadata buffer to be sufficiently sure about that one, but from what I remember, I believe that one actually uses m_Buffer as the actual buffer, and doesn't use m_Position.

Reply:
The BP3writer uses m_Position for both m_Metadata and m_Data sizes.  So updating both for consistency.  But you might be right about it not being necessary for m_Metadata.